File: src/models/semester.ts
export class Semester {
<<<<<<< HEAD
    constructor(
        public readonly semesterId: string,
        public readonly name: string,
        public readonly startDate: Date,
        public readonly endDate: Date,
        public isActive: boolean = false
    ) { }

    // Check if a given date falls within this semester
    containsDate(date: Date = new Date()): boolean {
        return date >= this.startDate && date <= this.endDate;
    }

    // Check if the semester is in the past
    isPast(): boolean {
        return new Date() > this.endDate;
    }

    // Check if the semester is in the future
    isFuture(): boolean {
        return new Date() < this.startDate;
    }

    // Get duration in weeks
    getDurationInWeeks(): number {
        const msPerWeek = 1000 * 60 * 60 * 24 * 7;
        return Math.ceil((this.endDate.getTime() - this.startDate.getTime()) / msPerWeek);
    }
=======
  constructor(
    public id: number,
    public name: string,
    public startDate: string, // "YYYY-MM-DD"
    public endDate: string // "YYYY-MM-DD"
  ) {}

  getId(): number {
    return this.id;
  }

  getName(): string {
    return this.name;
  }

  getStartDate(): string {
    return this.startDate;
  }

  getEndDate(): string {
    return this.endDate;
  }

  setName(name: string): void {
    this.name = name;
  }

  setStartDate(startDate: string): void {
    this.startDate = startDate;
  }

  setEndDate(endDate: string): void {
    this.endDate = endDate;
  }
>>>>>>> 3c2075c122738a12ae63f376a144582430dd3f95
}

File: Classes/EnrollEase/Semester.cpp
#include "Semester.h"

namespace EnrollEase {

    Semester::Semester(int id,
                       const std::string& name,
                       const std::string& startDate,
                       const std::string& endDate)
      : id(id),
        name(name),
        startDate(startDate),
        endDate(endDate)
    {}

    int Semester::getId() const {
        return id;
    }

    std::string Semester::getName() const {
        return name;
    }

    std::string Semester::getStartDate() const {
        return startDate;
    }

    std::string Semester::getEndDate() const {
        return endDate;
    }

    void Semester::setName(const std::string& n) {
        name = n;
    }

    void Semester::setStartDate(const std::string& sd) {
        startDate = sd;
    }

    void Semester::setEndDate(const std::string& ed) {
        endDate = ed;
    }

}
